### PR TITLE
Add comments and fix few todos in Metrics

### PR DESCRIPTION
**Pull request by cijothomas**
Towards #2621

### CODE REVIEW

File: test/Benchmarks/Metrics/MetricsBenchmarks.cs [hunk 1]
```diff
@@ -20,7 +20,6 @@
 using System.Diagnostics.Metrics;
 using BenchmarkDotNet.Attributes;
 using OpenTelemetry;
-using OpenTelemetry.Exporter;
```
Comment (reyang) [inline test/Benchmarks/Metrics/MetricsBenchmarks.cs:23]: Do we know why these "unused using statements" are not discovered by the CI?

**Review by reyang — APPROVED** [review]

File: src/OpenTelemetry/Metrics/ExplicitBucketHistogramConfiguration.cs [hunk 2]
```diff
@@ -24,6 +24,8 @@ public class ExplicitBucketHistogramConfiguration : MetricStreamConfiguration
         /// </summary>
         /// <remarks>
         /// The array must be in ascending order with distinct values.
+        /// Empty array means no histogram buckets will be generated.
```
Comment (utpilla) [inline src/OpenTelemetry/Metrics/ExplicitBucketHistogramConfiguration.cs:27]: Rephrase "means" -> "would result in"
Something like:

- An empty array would result in no histogram buckets being used. 
- Null value would result in default histogram buckets being used.

File: src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs [hunk 3]
```diff
@@ -33,6 +38,12 @@ public class PeriodicExportingMetricReader : BaseExportingMetricReader
         private readonly ManualResetEvent shutdownTrigger = new ManualResetEvent(false);
         private bool disposed;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PeriodicExportingMetricReader"/> class.
+        /// </summary>
+        /// <param name="exporter">Exporter instance to export Metrics to.</param>
+        /// <param name="exportIntervalMilliseconds">The interval in milliseconds between two consecutive exports. The default value is 60000.</param>
+        /// <param name="exportTimeoutMilliseconds">How long the export can run before it is cancelled. The default value is 30000.</param>
```
Comment (utpilla) [inline src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs:46]: nit: canceled*

File: test/Benchmarks/Metrics/MetricsBenchmarks.cs (hunk 1 above)
Comment (cijothomas) [reply test/Benchmarks/Metrics/MetricsBenchmarks.cs:23]: I think /test folder is not covered.

File: src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs (hunk 3 above)
Comment (cijothomas) [reply src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs:46]: will pass that.

**Review by utpilla — APPROVED** [review]

**Pull request merged**